src/Symfony/Component/Security/Core/User/User.php

```diff
@@ -11,12 +11,16 @@
 
 namespace Symfony\Component\Security\Core\User;
 
+trigger_deprecation('symfony/security-core', '5.3', 'The "%s" class is deprecated, use "%s" instead.', User::class, InMemoryUser::class);
```

Comment:
what about also telling that the class will be removed in v6.0?
according to the comment https://github.com/symfony/symfony/pull/40443/files#r592508122

Reply:
We don't do that anymore. Anything deprecated in 5.x is going to be removed in 6.0, so being explicit is not needed.

Reply:
Understood, I was seeing sometime that is why :)